Hook the CreateDirectory*() functions

TheComputerGuy96 opened the pull request:
ZMenu still doesn't fully work with modloader though (the strings for the `Mod Configuration` menu are all empty which doesn't happen when using a normal ASI loader)

Maybe that issue could be worked around by delaying the loading of the modloader strings (but the mod should probably be fixed itself with some memory patches?)

TheComputerGuy96 committed:
Hook the CreateDirectory*() functions

ZMenu calls CreateDirectoryW() to create the ZMenu directory for
storing some .ini files (which currently ends up in the root game folder)

thelink2012 commented:
Hey @TheComputerGuy96. Thanks for your contribution. I'll merge both this and #120 and publish a new release soon.

Just worried that this may cause some side-effect somewhere it shouldn't. But so far I've not thought of a edge case.

SandeMC commented:
@thelink2012 hey, any news on that release?

thelink2012 commented:
Hey @SandeMC, a bit busy here so I'll probably not be able to merge this before May.

CanerKaraca23 commented:
> Hey @SandeMC, a bit busy here so I'll probably not be able to merge this before May.

Hey, i hope you're not busy now.

CookiePLMonster reviewed (changes requested):
`CreateDirectoryExA` and `CreateDirectoryExW` also need shims.

thelink2012 commented:
@TheComputerGuy96 thanks and sorry for the long delay. 

Which mod is behaving badly because of the lack of CreateDirectory hooks? Can you provide more details? Is it ZMenu?

TheComputerGuy96 commented:
@thelink2012 ZMenu creates its own folder in the GTA San Andreas root folder (or maybe the folder relative to the .asi?) to store a speedrun configuration file (but due to the missing hook, the directory gets created in the wrong place and thus that config file likely can't be created due to the missing folder)

thelink2012 committed:
std.asi: skip translation when path argument is nullptr

Guards TranslatePathChar before IsAbsolutePath; optional API parameters may be NULL (e.g. CreateDirectoryEx template directory).

thelink2012 committed:
std.asi: use AR_PATH_IN for new dirs, INE for CreateDirectoryEx template

CreateDirectory/CreateDirectoryW and CreateDirectoryEx new-directory args use
IN; Ex lpTemplateDirectory uses INE when a template path is present.

thelink2012 commented:
Thanks @TheComputerGuy96!
